Add jetpack compatibility

Slaith12 opened the pull request:
Another simple implementation, I decided to make this separate from the grappling hook PR so that you could review/approve these separately (especially since this one might be harder to approve than the other one)

The recipes for each jetpack isn't quite equivalent to the original recipes, but I feel they still make sense for their tech levels and how expensive they originally were (although they are overall more expensive in this version)

Jetpack 1 is essentially equivalent to its original recipe. The only thing worth mentioning is that I replaced steel with sophisticated matter in the recipe, when originally all of its ingredients were pre-blue science. The jetpack was still a blue science tech though, and SMU is a fairly simple recipe, so I thought it still fit well.
Jetpack 2 has spectralite added on top of its other ingredients. I wanted to add an ingredient that you have to research and work towards before getting this jetpack, similar to how the electric engine works for bots (and this jetpack) in vanilla. Since the engine no longer fills that role due to being a blue science ingredient, I chose spectralite instead since I felt it fit the best out of the other options.
Jetpack 3 has arcane drive added. I don't really have as much justification for this as I do for jetpack 2, I just felt that the first 2 established a pattern of having a material/structural component, an engine, and a processor for each jetpack recipe. This also bumps the jetpack 3 up a tech level (yellow only -> purple + yellow), and is the change I like the least.
Jetpack 4 also continues the pattern mentions above, and it does so in a much more... interesting way. It's at the end-game and it was already hilariously expensive in vanilla, so I thought this was a fun way to integrate it into Ultracube.

Slaith12 committed:
Add jetpack recipes

Slaith12 committed:
Add jetpack fuels

Slaith12 committed:
Various fixes

It was pretty funny (and upsetting) seeing literally nothing work when I first tested it, but thankfully it was all just easy fixes

Slaith12 committed:
Merge remote-tracking branch 'upstream/main' into jetpack

grandseiken commented:
In the spirit of ultracube removing unnecessary clutter, how would you feel about only have 2/3 tiers instead of 4? I'm already kind of feeling like the current tiers 2/3 are too close together in the tech tree, so maybe we could just merge them.

I'm assuming it's intentional rather than an oversight that crafting the jetpack 4 softlocks your game?

Slaith12 commented:
I wouldn't be opposed to removing a tier. Personally I feel tiers 1/2 are too close, both being in blue science (which is also part of the reason I felt the need to add an ingredient to tier 2 to separate it more). Tier 3 seems to be a good distance away for me since it's in purple/yellow science.

As for tier 4, I'm pretty sure the recipe returns all the debris you need to avoid softlocking. 1 debris --> 2 matter, so 40 matter = 20 debris. The tier 4 recipe is supposed to be a challenge to craft, but it shouldn't be the last thing you ever craft.

grandseiken commented:
Ah, sorry, I only scanned and didn't realise the results were modified as well. Ok, yeah, either merging tier 1&2 or 2&3 gets to the same result in the end.

Slaith12 committed:
Adjusted jetpack tiers

The main thing I did here was a remove a tier so there are now 3 jetpack tiers, and the recipes are adjusted based on that.

Tier 1 is a combination of the old tiers 1 and 2 recipes, while tier 2 is essentially the old tier 3 recipe. Each of them has the same thrust as before, so they're basically just more expensive than they were previously.
Tier 3 is the old tier 4, even including the thrust value and a similar overall price. I also made the cube cost of this more expensive (it was actually like this in the first commit I did. I toned it down in the "various fixes" commit, but decided to revert it since I feel the composite is a more fitting ingredient next to the modules)
